### PR TITLE
Refactor FileUploadPathProvider to be more robust

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
1. Always append local scheme to URI if no scheme exists
2. Check whether all temp directory are local while constructing the class instance
3. Adding more logs for the location of all directories

### CODE REVIEW

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/FileUploadPathProvider.java [hunk 1]
```diff
@@ -30,79 +30,79 @@
...
+      mkdirIfNotExists(pinotFS, _schemasTmpDirURI);
+
+      // URIs that are always local
+      String localTempDir = controllerConf.getLocalTempDir();
+      if (localTempDir == null) {
+        LOGGER.info("Local temporary directory is not configured, use data directory as the local temporary directory");
+        _localTempDirURI = _baseDataDirURI;
       } else {
-        scheme = dataDirScheme;
+        _localTempDirURI = getUriFromPath(localTempDir);
```
Comment (jenniferdai) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/FileUploadPathProvider.java]: What input are you planning on allowing?
I see that normal directories are allowed but curious about different file:// prefixes? Do all amounts of slashes work?

Comment (Jackie-Jiang) [reply pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/FileUploadPathProvider.java]: None of them works except for file://. There is a check just after this

**Comment by jenniferdai** [comment]
In that case, please update docs/microsoft folks

**Comment by Jackie-Jiang** [comment]
@jenniferdai What do you mean? It's the same as current behavior, where local dir must use file://. The only difference is we allow address without scheme.

**Comment by Jackie-Jiang** [comment]
All "file" scheme will work

**Pull request merged**